transformer_engine/pytorch/csrc/extensions.cu

```diff
@@ -393,8 +393,226 @@ at::Tensor cast_from_fp8(const at::Tensor &input,
 }
 
 
+at::Tensor scaled_softmax_forward(at::Tensor input,
+                                  float scale_factor
+) {
+    using namespace transformer_engine;
+    AT_ASSERTM(input.dim() == 4, "expected 4D tensor");
+    AT_ASSERTM((input.scalar_type() == at::ScalarType::Half) ||
+               (input.scalar_type() == at::ScalarType::BFloat16),
+               "Only fp16 and bf16 are supported");
```

Comment:
This sucks - we should have some plan to lift that limitation at least in the followup PR.